Sources/Shared/DiffAware.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  DiffAware.swift
+//  DeepDiff
+//
+//  Created by khoa on 22/02/2019.
+//  Copyright © 2019 Khoa Pham. All rights reserved.
+//
+
+import Foundation
+
+/// Model must conform to DiffAware for diffing to work properly
+/// idProviding: To tell if there is deletion or insertion
+/// comparing: To tell if there is replacement or movement
+public protocol DiffAware {
+  var idProviding: Int { get }
```

Comment:
I'm a stickler for naming and this doesn't look like the best name. Why not just call it `id` or `uniqueId`?

Reply:
Yeah, I was trying ideas on non-protocol and bring the name here, I will use diffId instead